File: leetcode/moveZeros.js
var moveZeroes = function(nums) {
  let i = 0;
  let j = 0;
  for (; i < nums.length; i++) {
    if (nums[i] !== 0) {
      nums[j] = nums[i];
      j++;
    }
  }
  while (j < nums.length) {
    nums[j] = 0;
    j++;
  }
  return nums;
};

console.log(moveZeroes([1, 2, 0, 5, 3]));

File: python/EasyInterviewQuestions/arrays/moveZeros.py
class Solution:
    def moveZeroes(self, nums):
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        i = 0
        j = 0

        while j < n:
            if nums[j] != 0:
                nums[i] = nums[j]
                i += 1
            j += 1

        while i < n:
            nums[i] = 0
            i += 1

